mongo/gridfs/download_stream.go

```diff
@@ -119,6 +119,9 @@ func (ds *DownloadStream) Close() error {
 	}
 
 	ds.closed = true
+	if ds.cursor != nil {
+		return ds.cursor.Close(context.Background())
```

Comment:
No need to change. I was initially slightly concerned. This adds blocking work with `context.Background()`, which could block indefinitely. But previously this would always return immediately.

Though, if the cursor is exhausted, this should do no work (killCursors won't be sent to the server). If the cursor is not exhausted, this still seems preferable to leaking cursors server-side in the event of an error.

Reply:
A good point and valid concern. But, leaking cursors doesn't seem great; and on the driver side, not closing the cursor in this case will leave a session open (`client.NumberSessionsInProgress() > 0`).